pkg/frontend/lease_service.go

```diff
@@ -3,26 +3,285 @@ package frontend
 import (
 	"context"
 	"fmt"
+	"os"
+	"time"
+
+	klogv2 "k8s.io/klog/v2"
 
 	"go.etcd.io/etcd/etcdserver/etcdserverpb"
+
+	storageerrors "github.com/khenidak/london/pkg/backend/storageerrors"
+	"github.com/khenidak/london/pkg/types"
+)
+
+const (
+	leaseManagementElectionName = "lease-manager"
 )
 
+func (fe *frontend) leaseMangementLoop() {
+	var leader types.LeaderElect
+	myName, err := os.Hostname()
+	if err != nil {
+		// let us generate a random name
+		myName := fmt.Sprintf("%v", time.Now().UTC().UnixNano())
+		klogv2.Infof("lease manager: failed to get hostname with err:%v and used a random generated name instead:%v - process restart will yeild in lease loss", err, myName)
+	}
+	var ticker *time.Ticker
+	if fe.config.LeaseMgmtRunInterval != 0 {
+		klogv2.Infof("lease manager: operating at test 1s intervals")
+		ticker = time.NewTicker(time.Second * time.Duration(fe.config.LeaseMgmtRunInterval))
+	} else {
+		klogv2.Infof("lease manager: operating at 10s intervals")
+		ticker = time.NewTicker(time.Second * 10)
+	}
+
+	stopCh := fe.config.Runtime.Context.Done()
+	for {
+		select {
+		case <-stopCh:
+			// done here
+			return
+		case <-ticker.C:
```

Comment:
This ticker will not tick at t0. The first tick will only be at t0+10. Wouldn't that cause unwanted delay here? We should force a tick before entering the loop.

Reply:
This would be fine. The first run is delayed.The clean up is best efforts really.